qualtran/bloqs/chemistry/trotter/trotterized_unitary.py

```diff
@@ -0,0 +1,121 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Bloq for building a Trotterized unitary"""
+
+from functools import cached_property
+from typing import Dict, Iterable, Union
+
+import attrs
+import sympy
+
+from qualtran import Bloq, bloq_example, BloqBuilder, BloqDocSpec, Signature, Soquet, SoquetT
+
+
+@attrs.frozen
+class TrotterizedUnitary(Bloq):
+    r"""Implement arbitrary trotterized unitary given any Trotter splitting of the Hamiltonian.
+
+    Given an arbitrary splitting of the Hamiltonian into $\Gamma$ terms
+
+    $$
+        H = \sum_{\gamma=1}^\Gamma H_\gamma
+    $$
+
+    then the unitary $e^{-i t H}$ can be approximately implemented via a $p$-th order product
+    formula
+
+    $$
+        S_p(t) = \prod_{v=1}^{\Upsilon}\prod_{k=1}^\Gamma e^{-it a_{v,\gamma} H_{\pi_v(\gamma)}}
```

Comment:
if it was a typo, I say we point it out somewhere in our docs

Reply:
I think the reference is a bit more general (so t can be real or imaginary) to encompass both unitary evolution and imaginary time evolution since trotterization is used for both cases. It just looked a little weird to me to omit the the $-i$ for us.